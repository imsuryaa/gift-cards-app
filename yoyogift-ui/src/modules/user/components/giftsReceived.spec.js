import React from "react";
import { mount } from "enzyme";
import GiftsReceived from "./giftsReceived";

describe("Gifts Received Test Suite", () => {
  it("should render the expected HTML", () => {
    const classes = {
      root: { marginTop: "2px" },
      table: { marginTop: "2px" }
    };

    expect(mount(<GiftsReceived 
      classes={classes}
      data={[]}/>
      ).html()).toMatchSnapshot();
  });

  it("should render the expected HTML", () => {
    const classes = {
      root: { marginTop: "2px" },
      table: { marginTop: "2px" }
    };
    const data = [
      {
        cardName: 'A',
        cardPoints: 1,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: true
      },
      {
        cardName: 'B',
        cardPoints: 2,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      },
      {
        cardName: 'C',
        cardPoints: 3,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      }
    ];
    expect(mount(<GiftsReceived 
      classes={classes}
      data={data}/>
      ).html()).toMatchSnapshot();
  });

  it("should find only table heading", () => {
    const classes = {
      root: { marginTop: "2px" },
      table: { marginTop: "2px" }
    };

    const wrapper = mount(<GiftsReceived 
      classes={classes}
      data={[]}/>
      );

    expect(wrapper.find('th').length).toBe(7);
    expect(wrapper.find('td').length).toBe(0);
  });

  it("should find table heading and 3 rows", () => {
    const classes = {
      root: { marginTop: "2px" },
      table: { marginTop: "2px" }
    };
    const data = [
      {
        cardName: 'A',
        cardPoints: 1,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: true
      },
      {
        cardName: 'B',
        cardPoints: 2,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      },
      {
        cardName: 'C',
        cardPoints: 3,
        senderEmail: 'knowledge.com2k16@gmail.com',
        cardIssueDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        cardExpiryDate: "Sun May 19 2019 15:43:25 GMT+0530 (India Standard Time)",
        isRedeemed: false
      }
    ];

    const wrapper = mount(<GiftsReceived 
      classes={classes}
      data={data}/>
      );

    expect(wrapper.find('tr').length).toBe(4);
    expect(wrapper.find('th').length).toBe(10);
    expect(wrapper.find('td').length).toBe(18);
  });
});
