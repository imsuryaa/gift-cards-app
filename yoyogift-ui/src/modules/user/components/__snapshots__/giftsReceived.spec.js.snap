// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts Received Test Suite should render the expected HTML 1`] = `"<div class=\\"MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 GiftsReceived-root-1 [object Object]\\"><table class=\\"MuiTable-root-30 GiftsReceived-table-2 [object Object]\\"><thead class=\\"MuiTableHead-root-31\\"><tr class=\\"MuiTableRow-root-32 MuiTableRow-head-35\\"><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">CARD NAME</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">POINTS</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">RECEIVED FROM</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">CARD DESC</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">ISSUE DATE</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">EXPIRY DATE</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">REDEEM</th></tr></thead><tbody class=\\"MuiTableBody-root-49\\"></tbody></table></div>"`;

exports[`Gifts Received Test Suite should render the expected HTML 2`] = `"<div class=\\"MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 GiftsReceived-root-1 [object Object]\\"><table class=\\"MuiTable-root-30 GiftsReceived-table-2 [object Object]\\"><thead class=\\"MuiTableHead-root-31\\"><tr class=\\"MuiTableRow-root-32 MuiTableRow-head-35\\"><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">CARD NAME</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">POINTS</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">RECEIVED FROM</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">CARD DESC</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">ISSUE DATE</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">EXPIRY DATE</th><th class=\\"MuiTableCell-root-37 MuiTableCell-head-38\\" scope=\\"col\\">REDEEM</th></tr></thead><tbody class=\\"MuiTableBody-root-49\\"><tr class=\\"MuiTableRow-root-32\\"><th class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\" scope=\\"row\\">A</th><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">1</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">knowledge.com2k16@gmail.com</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\"></td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">Redeemed</td></tr><tr class=\\"MuiTableRow-root-32\\"><th class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\" scope=\\"row\\">B</th><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">knowledge.com2k16@gmail.com</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\"></td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\"><button class=\\"MuiButtonBase-root-76 MuiButton-root-50 MuiButton-contained-61 MuiButton-containedPrimary-62 MuiButton-raised-64 MuiButton-raisedPrimary-65\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-51\\">Redeem</span><span class=\\"MuiTouchRipple-root-79\\"></span></button></td></tr><tr class=\\"MuiTableRow-root-32\\"><th class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\" scope=\\"row\\">C</th><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">3</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">knowledge.com2k16@gmail.com</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\"></td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\">2019-05-19</td><td class=\\"MuiTableCell-root-37 MuiTableCell-body-39\\"><button class=\\"MuiButtonBase-root-76 MuiButton-root-50 MuiButton-contained-61 MuiButton-containedPrimary-62 MuiButton-raised-64 MuiButton-raisedPrimary-65\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-51\\">Redeem</span><span class=\\"MuiTouchRipple-root-79\\"></span></button></td></tr></tbody></table></div>"`;
